src/ros_filter.cpp

```diff
@@ -2100,7 +2100,18 @@ namespace RobotLocalization
         tf2::Vector3 normAcc(0, 0, 9.80665);
         tf2::Quaternion curAttitude;
         tf2::Transform trans;
-        tf2::fromMsg(msg->orientation, curAttitude);
+
+        if(::fabs(msg->orientation_covariance[0] + 1) < 1e-9)
```

Comment:
I like this idea, but this assumes that the IMU and the robot are not offset from each other rotationally (i.e., that you IMU rests level on the robot, and not on its side or upside-down). Really, you need to rotate the state's RPY value back into the IMU's coordinate frame, and then apply this logic.